### PR TITLE
ISSUE-196: Enable TLS on http2 adaptors.

### PR DESCRIPTION
**Pull request by ganeshmurthy**
Support for TLSv1.2 and TLSv1.3
ALPN support where available

### CODE REVIEW

**Review by jiridanek — CHANGES_REQUESTED** [review]
I think you are missing `-DBUILD_TLS=ON` for Proton build in `.github/scripts/compile.sh`.

**Comment by ganeshmurthy** [comment]
> I think you are missing `-DBUILD_TLS=ON` for Proton build in `.github/scripts/compile.sh`.

Thanks. I added this

**Comment by jiridanek** [comment]
https://github.com/skupperproject/skupper-router/runs/5851875396?check_suite_focus=true#step:26:2902

(I figure you already know, but just to be sure.)

```
56: ERROR: Aborted due to unexpected alloc pool leak of type 'qdr_link_t'
56: 
56: <<<<
56: 
56: Router QDR debug dump file:
56: >>>>
56: alloc.c: Items of type 'qd_iterator_t' remain allocated at shutdown: 1 (SUPPRESSED)
56: alloc.c: Items of type 'qdr_field_t' remain allocated at shutdown: 1 (SUPPRESSED)
56: alloc.c: Items of type 'qd_buffer_t' remain allocated at shutdown: 1 (SUPPRESSED)
56: alloc.c: Items of type 'qdr_terminus_t' remain allocated at shutdown: 2
56: Leak: 2022-04-06 13:48:04.996099 +0000 type: qdr_terminus_t address: 0x612000035f50
56: /lib64/libasan.so.6(+0x44701) [0x7fba53ea6701]
56: skrouterd(qd_alloc+0x1fe) [0x4c064e]
56: skrouterd(qdr_terminus+0x2e) [0x6226ae]
56: skrouterd() [0x6c6f67]
56: skrouterd() [0x6d27f2]
56: skrouterd() [0x67aa85]
56: skrouterd() [0x682156]
56: /lib64/libc.so.6(+0x8db1a) [0x7fba528b4b1a]
56: /lib64/libc.so.6(+0x112660) [0x7fba5[2939](https://github.com/skupperproject/skupper-router/runs/5851875396?check_suite_focus=true#step:26:2939)660]
56: 
56: Leak: 2022-04-06 13:48:04.996123 +0000 type: qdr_terminus_t address: 0x612000035dd0
56: /lib64/libasan.so.6(+0x44701) [0x7fba53ea6701]
56: skrouterd(qd_alloc+0x1fe) [0x4c064e]
56: skrouterd(qdr_terminus+0x2e) [0x6226ae]
56: skrouterd() [0x6c7009]
56: skrouterd() [0x6d27f2]
56: skrouterd() [0x67aa85]
56: skrouterd() [0x682156]
56: /lib64/libc.so.6(+0x8db1a) [0x7fba528b4b1a]
56: /lib64/libc.so.6(+0x112660) [0x7fba52939660]
56: 
56: alloc.c: Items of type 'qdr_link_t' remain allocated at shutdown: 1
56: Leak: 2022-04-06 13:48:04.996134 +0000 type: qdr_link_t address: 0x617000085490
56: /lib64/libasan.so.6(+0x44701) [0x7fba53ea6701]
56: skrouterd(qd_alloc+0x1fe) [0x4c064e]
56: skrouterd(qdr_link_first_attach+0x42) [0x5a4332]
56: skrouterd() [0x6c7047]
56: skrouterd() [0x6d27f2]
56: skrouterd() [0x67aa85]
56: skrouterd() [0x682156]
56: /lib64/libc.so.6(+0x8db1a) [0x7fba528b4b1a]
56: /lib64/libc.so.6(+0x112660) [0x7fba52939660]
56: 
56: alloc.c: Items of type 'qd_http2_buffer_t' remain allocated at shutdown: 1 (SUPPRESSED)
```

File: src/router_core/connections.c [hunk 1]
```diff
@@ -1702,16 +1708,24 @@ static void qdr_link_inbound_first_attach_CT(qdr_core_t *core, qdr_action_t *act
...
+    qdr_terminus_t *target = action->args.connection.target;
+
     if (discard || !conn || !link) {
         if (initial_dlv)
             qdr_delivery_decref(core, initial_dlv,
                                 "qdr_link_inbound_first_attach_CT - discarding action");
+        if (discard) {
+            qdr_terminus_free(source);
+            qdr_terminus_free(target);
+            qdr_link_free_CT(link);
```
Comment (kgiusti) [inline src/router_core/connections.c:1722]: Freeing the link doesn't seem safe to me: the callers to qdr_link_inbound_first_attach() get a pointer to the link returned to them, so I'm pretty sure freeing this link here will result in... issues.

Comment (ganeshmurthy) [reply src/router_core/connections.c:1722]: Agreed. Ideally I would like to drop this link into core->open_links so the shutdown process will clean it up. But it looks like the qdr_core_free() first cleans up all actions and then later cleans up  core->open_links. Sigh.

File: CMakeLists.txt [hunk 2]
```diff
@@ -73,7 +73,7 @@ find_library(rt_lib rt)
 
 find_package(Python 3.6.8 REQUIRED COMPONENTS Interpreter Development)
 find_package(Threads REQUIRED)
-find_package(Proton 0.34.0 REQUIRED COMPONENTS Core Proactor)
+find_package(Proton 0.37.0 REQUIRED COMPONENTS Core Proactor Tls)
```
Comment (jiridanek) [inline CMakeLists.txt:76]: I'm opening https://issues.apache.org/jira/browse/PROTON-2532, CMake find_package will fail unhelpfully when user asks for component that is not provided

See the RPM CI job failure to see how unhelpful the failure is.

**Review by kgiusti — APPROVED** [review]

**Pull request merged**